remove windows desktop certs from client keyring

nklaassen opened the pull request:
These aren't actually used in the client keyring anywhere, and clients never actually get desktop certs. These are the certs used for the desktop agent to dial to the windows desktop. They're only here for a `tctl auth sign` debug command just to pass the cert der through to the file writer. When we actually generate these desktop certs in lib/srv/desktop we don't use the client keyring at all.

I also removed `client.ReissueParams.RouteToWindowsDesktop` which was never written to anywhere

nklaassen committed:
remove windows desktop certs from client keyring

zmb3 commented on lib/client/identityfile/identity.go:
```diff
@@ -182,6 +182,8 @@ type WriteConfig struct {
 	OutputPath string
 	// KeyRing contains the credentials to write to the identity file.
 	KeyRing *client.KeyRing
+	// WindowsDesktopCert contains windows desktop certs to write.
```
```suggestion
	// WindowsDesktopCerts contains windows desktop certs to write.
```

espadolini commented on lib/client/client.go:
```diff
@@ -184,8 +181,6 @@ func (p ReissueParams) isMFARequiredRequest(sshLogin string) *proto.IsMFARequire
 		req.Target = &proto.IsMFARequiredRequest_KubernetesCluster{KubernetesCluster: p.KubernetesCluster}
 	case p.RouteToDatabase.ServiceName != "":
 		req.Target = &proto.IsMFARequiredRequest_Database{Database: &p.RouteToDatabase}
-	case p.RouteToWindowsDesktop.WindowsDesktop != "":
-		req.Target = &proto.IsMFARequiredRequest_WindowsDesktop{WindowsDesktop: &p.RouteToWindowsDesktop}
```
Should we add a default case here? Should've we always had a default case here?

nklaassen replied:
I don't think we ever end up here or else it's a logic bug, but that's not obvious at all, so added a default case that returns an error

nklaassen committed:
fix typo in comment

Co-authored-by: Zac Bergquist <zac.bergquist@goteleport.com>

nklaassen committed:
add default case

nklaassen committed:
fix test

nothing actually looks at the mfa required response if the error is
non-nil, the assertion was pointless and only testing an implementation
detail of the test harness

nklaassen committed:
remove rsa keygen from identity file test
